web/src/reducers/dataReducer.js

```diff
@@ -96,7 +115,10 @@ module.exports = (state = initialDataState, action) => {
         newGrid.zones[key] = zone;
       });
       Object.keys(newGrid.exchanges).forEach((key) => {
-        newGrid.exchanges[key].netFlow = undefined;
+        const exchange = Object.assign({}, newGrid.exchanges[key]);
```

Comment:
Good catch, these can be very dangerous! :grimacing: 

By the way, in ES6 this can be written idiomatically as `const exchange = { ...newGrid.exchanges[key] };`